perun-base/src/main/java/cz/metacentrum/perun/core/api/exceptions/AttributeNotExistsException.java

```diff
@@ -2,21 +2,35 @@
 
 
 /**
- * Attribute not exists in underlaying data source.
+ * This exception is thrown when the attribute definition does not exist in the database. However, if the attribute definition exists
+ * but isn't just set for the given entity, this exception does not raise.
```

Comment:
`This exception is thrown when the attribute does not exist in underlying data source.`, which is quite similar to what you suggest, was what i wrote first but Slávek wanted me to change it so that it contains the info which it does now. If you want me to change it anyway, no problem.

Reply:
@licehammer What do you think?

Reply:
I prefer the longer explanation, because it's something which is often misunderstood by people using the API. 
I would even expand the sentence like this:
`However, if the attribute definition exists but the attribute values is not set for the given entity, this exception does not raise and the situation is usually resolved by returning Attribute object with null value instead.`

Reply:
@licehammer I agree.